src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -198,6 +198,40 @@
 
 
 ;; ## Secret Usage
+(defn- lookup-paths
+  "Creates a map from Vault paths (as keywords) to their corresponding secret data (defaults to nil)
+
+  Params:
+  - `client`: The vault.client you want to use to access Vault
+  - `paths`: A seq containing all paths you wish to look up (as strings or keywords), duplicates are fine."
+  [client paths]
+  (letfn [;; Makes a vector containing the path and the data associated with that path in vault, or nil if none is found
+          (lookup-path
+            [path]
+            [(keyword path) (vault/read-secret client path {:not-found nil})])]
```

Comment:
Yeah, that was the idea. Just makes it a bit easier to access. Right now, it's being accessed below in `(-> paths-to-vals path key)` which would have to otherwise be: `((keyword key) (get paths-to-vals path))`